client/src/pages/ProfileListings/ProfileListings.tsx

```diff
@@ -0,0 +1,120 @@
+  const [search, setSearch] = useState<string>('');
+
+  const moment = extendMoment(Moment);
+
+  const handleShowMore = () => {
+    const numberOfUsers = displayedUsers.length;
+    const lastUser = users[displayedUsers.length];
+    if (lastUser === undefined) {
+      setSnackbarOpen(true);
+    }
+    setDisplayedUsers(users.slice(0, numberOfUsers + 3));
+  };
+
+  const handleSnackbarClose = (event?: React.SyntheticEvent, reason?: string) => {
+    if (reason === 'clickaway') {
+      return;
+    }
+    setSnackbarOpen(false);
+  };
+
+  const handleCalendarOpen = () => {
+    setCalendarOpen((prevState) => !prevState);
+  };
+
+  const updateDateRange = (dateFrom: Date | null, dateTo: Date | null) => {
+    if (dateFrom instanceof Date && dateTo instanceof Date) {
+      setDateRange({
+        from: dateFrom,
+        to: dateTo,
+      });
+      const formattedDateFrom = moment(dateFrom).format('MMMM Do YYYY');
+      const formattedDateTo = moment(dateTo).format('MMMM Do YYYY');
+      setFormattedDateRange(`${formattedDateFrom} – ${formattedDateTo}`);
+    } else {
+      setFormattedDateRange(`any`);
+    }
+  };
+
+  const profileCardGrid = (
+    <Grid container className={classes.profilesContainer} xs={9}>
+      {displayedUsers.map((user) => (
+        <ProfileCard user={user} key={user._id} />
+      ))}
+      <Snackbar open={snackbarOpen} autoHideDuration={2000} onClose={handleSnackbarClose}>
+        <Alert severity="info">No more users found</Alert>
+      </Snackbar>
+    </Grid>
+  );
+
+  return (
+    <Grid container className={classes.root}>
+      <Typography variant="h4" component="h1" className={classes.title}>
+        Your search results
+      </Typography>
+      <Grid>
+        <Grid container className={classes.searchDateContainer}>
+          <Grid className={classes.search}>
+            <SearchIcon className={classes.searchIcon} />
+            <InputBase
+              placeholder="Search by city…"
+              inputProps={{ 'aria-label': 'search' }}
+              onChange={(e) => {
+                setSearch(e.target.value);
+              }}
+            />
+          </Grid>
+          <Grid>
+            <Divider orientation="vertical" />
+          </Grid>
+          <Grid className={classes.dateRangeContainer}>
+            <Button onClick={handleCalendarOpen}>
+              <DateRangeIcon className={classes.dateRangeIcon} />
+            </Button>
+            <Typography>{formattedDateRange.charAt(0).toUpperCase() + formattedDateRange.slice(1)}</Typography>
+            <Button onClick={() => setFormattedDateRange('any')}>
+              <CloseIcon className={classes.dateRangeIcon} />
+            </Button>
+          </Grid>
+        </Grid>
+      </Grid>
+      {profileCardGrid}
+      {search === '' && formattedDateRange === 'any' ? (
+        <Button onClick={handleShowMore} variant="outlined" className="showMore">
+          Show More
+        </Button>
+      ) : null}
```

Comment:
Whenever you only have an `if` but no `else`, use this pattern instead:
```javascript
{search === '' && formattedDateRange === 'any' && (
    <Button onClick={handleShowMore} variant="outlined" className="showMore">Show More</Button>
)}
```
Also can we do `!search` here?